pymmcore_plus/core/_property.py

```diff
@@ -0,0 +1,145 @@
+from typing import Any, Optional, Tuple
+
+from pymmcore import g_Keyword_Label, g_Keyword_State
+
+from ._constants import DeviceType, PropertyType
+from ._mmcore_plus import CMMCorePlus
+
+
+class MMProperty:
+    """Convenience "View" onto a device property.
+
+    Parameters
+    ----------
+    device_label : str
+        Device this property belongs to
+    property_name : str
+        Name of this property
+    mmcore : Optional[CMMCorePlus]
+        CMMCore instance, by default global singleton.
+
+    Examples
+    --------
+
+    >>> prop = MMProperty('Objective', 'Label')
+    >>> prop.dict()
+    >>> prop.core.loadSystemConfiguration()
+    >>> prop.dict()
+    >>> prop.value
+    >>> prop.value = 'Objective-2'
+    >>> prop.isReadOnly()
+    >>> prop.hasLimits()
+    >>> prop.range()
+    ...
+    """
+
+    def __init__(
+        self,
+        device_label: str,
+        property_name: str,
+        *,
+        mmcore: Optional[CMMCorePlus] = None,
+    ) -> None:
+
+        self.device = device_label
+        self.name = property_name
+        self._mmc = mmcore or CMMCorePlus.instance()
+
+    def isValid(self) -> bool:
+        return self.isLoaded() and self._mmc.hasProperty(self.device, self.name)
+
+    def isLoaded(self) -> bool:
+        return self._mmc is not None and self.device in self._mmc.getLoadedDevices()
+
+    @property
+    def core(self) -> CMMCorePlus:
+        return self._mmc
+
+    @core.setter
+    def core(self, mmcore: CMMCorePlus) -> None:
+        self._mmc = mmcore
+
+    # functional alternate to property setter
+    def setCore(self, mmcore: CMMCorePlus) -> None:
+        self._mmc = mmcore
```

Comment:
> Most importantly, if I get a prop from one core, switch it's value, then swap it to another does that new core get sent a setProperty

nope, this important thing to recognize here is that this is a completely stateless class (outside of core, device label and property name).  It simply wraps the api of `core.doAThin(device, property)` ... so if you switch core, you switch what you would get from `core.getAthing(device, property)`.

Reply:
(that's really what I meant by "view" here... Perhaps calling this a `PropertyView` would be good?)

Reply:
>  nope, this important thing to recognize here is that this is a completely stateless class 

Ah yes - i wasn't considering that! So i guess my worry is more so that it may be easy to assume that these are stateful object (like i did) and then be surprised when they are not.

Also I hope that people don't have multiple cores lying around, that seems like it would only lead to bad times.

Reply:
yeah, fair enough :)  OK, going to make core a mandatory parameter in `__init__` and unsetable

Reply:
(if we ever later decide that it would be nice to instantiate one of these objects before you even know about a core, we can change later)

Reply:
Don't forget the middle ground of `None` and defaulting to using `instance` potentially avoids some typing in the hopefully common case of only one core. But I also see how it opens up a box of worms

Reply:
I think I'll add a method to CMMCorePlus ... so it would be CMMCorePlus.instance().propertyObject('dev', 'label')

Reply:
> CMMCorePlus.instance().propertyObject('dev', 'label')

What do you think about `getProperty` returning one of these objects?

Reply:
Ahh actually I see how that breaks the symmetry of the `get` and `set` property functions, which would be bad.

Reply:
I considered it, but I think that's too much of a divergence from the core API.  There are other cases, like `getConfigData` or other methods that return a `Configuration` where I added the `native=False` parameter, in which case we return an improved object... but those objects simply _add_ to the pymmcore API.   Here, we'd be fully changing the API from something that returns a value, to something that returns something that has a `.value` attribute ... so i think it's too confusing

Reply:
not married to `propertyObject` as the method name though!

Reply:
Definitely agree that it would become too confusing. 

I think this is pretty good. Only other ones I might consider are `getPropertyObject` or `getPropertyAsObject` for the similarity (but still important difference) with the core methods.

Reply:
I like `getPropertyObject` ... let's go with that